src/app/signup/page.tsx

```diff
@@ -50,74 +68,141 @@ const SignUp = () => {
           </div>
           <div className="flex items-center px-12 mb-10 w-full">
             <form
-              // @ts-expect-error Type mismatch
-              onSubmit={handleSubmit((data) => signupUser(data))}
+              ref={formRef}
+              onSubmit={handleSubmit(async (data) => {
+                const error = await signupUser(data);
+
+                if (error?.email?.length) {
+                  setEmailBeforeSubmission(data.email);
+                }
+
+                setServerErrors((prevError) => {
+                  if (!prevError) {
+                    return error;
+                  } else {
+                    return {
+                      ...prevError,
+                      ...error,
+                    };
+                  }
+                });
+              })}
               className="w-full"
+              noValidate
             >
               <div className="mb-4">
                 <input
                   autoFocus
+                  autoComplete="given-name"
```

Comment:
Are these names arbitrary? What is "given-name"?

Reply:
It's like the "real name". It works on my browser (Autocompleting names I have used)

Reply:
Is there a list of the supported names somewhere?

Reply:
To be honest, I don't know. I only know that it silenced the warnings and it made autocomplete work for the name and email. I'll remove it for now if it needs to be a sure addition.

Reply:
Ah. I found it https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete

That's so interesting and cool.

This looks good to me then, as long as we are following the spec here.

Reply:
Noted